src/hotspot/share/gc/shared/gcConfig.cpp

```diff
@@ -165,6 +198,10 @@ GCArguments* GCConfig::select_gc() {
   // Exactly one GC selected
   FOR_EACH_INCLUDED_GC(gc) {
     if (gc->_flag) {
+      _gc_name = gc->_hs_err_name;
+      if (ZGenerational) {
+        _gc_name = "z gen gc";
```

Comment:
Given that GenZGC will be the only ZGC option in the future, I don't think this needs to change for a potential upstream change.

But it is important in case this idea is ever backported to older JDKs.